tests/test_modeling_flax_roformer.py

```diff
@@ -0,0 +1,142 @@
+        self.hidden_size = hidden_size
+        self.num_hidden_layers = num_hidden_layers
+        self.num_attention_heads = num_attention_heads
+        self.intermediate_size = intermediate_size
+        self.hidden_act = hidden_act
+        self.hidden_dropout_prob = hidden_dropout_prob
+        self.attention_probs_dropout_prob = attention_probs_dropout_prob
+        self.max_position_embeddings = max_position_embeddings
+        self.type_vocab_size = type_vocab_size
+        self.type_sequence_label_size = type_sequence_label_size
+        self.initializer_range = initializer_range
+        self.num_choices = num_choices
+
+    def prepare_config_and_inputs(self):
+        input_ids = ids_tensor([self.batch_size, self.seq_length], self.vocab_size)
+
+        attention_mask = None
+        if self.use_attention_mask:
+            attention_mask = random_attention_mask([self.batch_size, self.seq_length])
+
+        token_type_ids = None
+        if self.use_token_type_ids:
+            token_type_ids = ids_tensor([self.batch_size, self.seq_length], self.type_vocab_size)
+
+        config = RoFormerConfig(
+            vocab_size=self.vocab_size,
+            hidden_size=self.hidden_size,
+            num_hidden_layers=self.num_hidden_layers,
+            num_attention_heads=self.num_attention_heads,
+            intermediate_size=self.intermediate_size,
+            hidden_act=self.hidden_act,
+            hidden_dropout_prob=self.hidden_dropout_prob,
+            attention_probs_dropout_prob=self.attention_probs_dropout_prob,
+            max_position_embeddings=self.max_position_embeddings,
+            type_vocab_size=self.type_vocab_size,
+            is_decoder=False,
+            initializer_range=self.initializer_range,
+        )
+
+        return config, input_ids, token_type_ids, attention_mask
+
+    def prepare_config_and_inputs_for_common(self):
+        config_and_inputs = self.prepare_config_and_inputs()
+        config, input_ids, token_type_ids, attention_mask = config_and_inputs
+        inputs_dict = {"input_ids": input_ids, "token_type_ids": token_type_ids, "attention_mask": attention_mask}
+        return config, inputs_dict
+
+
+@require_flax
+class FlaxRoFormerModelTest(FlaxModelTesterMixin, unittest.TestCase):
+
+    test_head_masking = True
+
+    all_model_classes = (
+        (
+            FlaxRoFormerModel,
+            FlaxRoFormerForMaskedLM,
+            FlaxRoFormerForSequenceClassification,
+            FlaxRoFormerForTokenClassification,
+            FlaxRoFormerForMultipleChoice,
+            FlaxRoFormerForQuestionAnswering,
+        )
+        if is_flax_available()
+        else ()
+    )
+
+    def setUp(self):
+        self.model_tester = FlaxRoFormerModelTester(self)
+
+    @slow
+    def test_model_from_pretrained(self):
+        for model_class_name in self.all_model_classes:
+            model = model_class_name.from_pretrained("junnyu/roformer_chinese_small", from_pt=True)
```

Comment:
This test requires PyTorch at the moment. @junnyu would it be ok if we upload flax checkpoints to your model repo: `"junnyu/roformer_chinese_small"`? It would make the user experience for using roformer in Flax much nicer with no downside really. The only small downside would be that if one wants to clone your repo it'll be a bit heavier

Reply:
@stancld - could we maybe also add one integration tests such as: https://github.com/huggingface/transformers/blob/8f6373c61ca71d2d8176e181b587a2307d8358c8/tests/test_modeling_roformer.py#L456 here?

Reply:
@patrickvonplaten OK! :)

Reply:
Awesome thank you :-) Will upload the model then

Reply:
Actually @stancld could you in a first step just create a `roformer_chinese_small` checkpoint under your namespace and then I'll transfer the weights in a second step once the test passes?

Reply:
@patrickvonplaten I've uploaded flax checkpoints here:

- https://huggingface.co/Stancld/roformer_chinese_small
- https://huggingface.co/Stancld/roformer_chinese_base
- https://huggingface.co/Stancld/roformer_chinese_char_small
- https://huggingface.co/Stancld/roformer_chinese_char_base

I compared `FlaxRoFormerForMaskedLM` loaded from the original `PT` checkpoint and from these new flax ones and the outputs seem to be the same.

Reply:
Epic uploaded your checkpoints to the official ones. Thanks a lot @stancld !